README.md

```diff
@@ -15,7 +18,96 @@ Sciences at UiT - The Arctic University of Norway.
- 
+
+### Versioning scheme
+
+This project adheres to a [semantic versioning scheme](https://semver.org/),
+with `MAJOR`.`MINOR`.`PATCH`-`LABEL`
+
+- `MAJOR`: Introduces backward incompatible changes to the API
+- `MINOR`: Introduces new features without breaking backward compatibility
+- `PATCH`: Only simple bugfixes
+- `LABEL`: Marks a pre-released version or release candidate
+
+We will try to keep only a *single* line of development on the `master`
+branch, and from this branch off to tag new releases. The type of release
+(`MAJOR` or `MINOR`) is determined by the nature of the introduced changes.
+`MAJOR` changes should not be introduced lightly, and could be held back at pull
+request-level in anticipation of prior `MINOR` release(s). Once a new `MAJOR` is
+released, the development on the old `MAJOR` is terminated and no `MINOR` pull
+requests will be accepted on top of old `MAJOR`s, only simple `PATCH`es.
+
+### Branching model
+
+In the following, X, Y and Z should *always* be substituted with appropriate
+numeric values, they should never appear as 'X', 'Y' or 'Z' in any version or
+branch name.
+
+The upstream MRChemSoft/mrcpp repository should contain only two kinds of
+branches: the `master` branch which represents the main development line, as
+well as a separate `release/X.Y` branch for each `MINOR` release. All `PATCH`
+releases are applied linearly on the `MINOR` release branches.
+
+#### The `master` branch:
+
+<img src="doc/gfx/git-master.png" alt="drawing" width="600"/>
+
+- This branch should *not* carry any release tags
+- New features should *always* be directed to this branch
+- Bugfixes may be directed to this branch
+- Bugfixes may be *cherry-picked* from `release` branches, but *never* merged
+- The VERSION file should point to the next *expected* release,
+  and *always* carry the pre-release label `-alpha`
+- When a new `release/X.Y` branch is created, the VERSION file on `master`
+  is bumped to the next expected `MAJOR`/`MINOR`, e.i. `X.(Y+1).0-alpha` or
+  `(X+1).0.0-alpha`
+
+#### The `release/X.Y` branches:
+
+<img src="doc/gfx/git-release.png" alt="drawing" width="530"/>
+
+- This branch should carry *all* release tags associated with the `X.Y` `MINOR`
+  release, including `PATCH`es and release candidates `-alpha1`, `-alpha2`, etc
+- The VERSION file should point to the *latest* release tag on this branch
+- The creation of this branch marks a feature freeze for version `X.Y`
+- New features should *never* be directed to this branch
+- Bugfixes and release preparations may be directed to this branch
+- Bugfixes may be *cherry-picked* from `master`, but *never* merged
+
+### Contributing
+
+All code changes are incorporated through the `fork -> pull request (PR) ->
+code review` work flow, with the following principles:
+
+- New features should *always* branch off `master` and PR back to `master`
+- Bugfixes may branch off either `master` or `release/X.Y`, but should *always*
+  PR back to its origin
```

Comment:
Meaning that they should the target of the PR is whatever base we branched off from? That is, PRs originating from branches off `master` should be directed back to `master`?

Reply:
Thanks, just realized it can be interpreted as "all derivative work *must* be PR-ed back to the main repository", which of course is not the case. I made a separate point of directing PRs.